pkg/cloud/services/loadbalancer/service.go

```diff
@@ -22,39 +22,39 @@ import (
 	"github.com/go-logr/logr"
 	"github.com/gophercloud/gophercloud"
 	"github.com/gophercloud/gophercloud/openstack"
-	"github.com/gophercloud/utils/openstack/clientconfig"
 
 	"sigs.k8s.io/cluster-api-provider-openstack/pkg/cloud/services/networking"
+	"sigs.k8s.io/cluster-api-provider-openstack/pkg/scope"
 )
 
 // Service interfaces with the OpenStack Neutron LBaaS v2 API.
 type Service struct {
 	projectID          string
+	scope              *scope.Scope
 	loadbalancerClient LbClient
 	networkingService  *networking.Service
-	logger             logr.Logger
 }
 
 // NewService returns an instance of the loadbalancer service.
-func NewService(client *gophercloud.ProviderClient, clientOpts *clientconfig.ClientOpts, logger logr.Logger) (*Service, error) {
-	loadbalancerClient, err := openstack.NewLoadBalancerV2(client, gophercloud.EndpointOpts{
-		Region: clientOpts.RegionName,
+func NewService(scope *scope.Scope) (*Service, error) {
+	loadbalancerClient, err := openstack.NewLoadBalancerV2(scope.ProviderClient, gophercloud.EndpointOpts{
+		Region: scope.ProviderClientOpts.RegionName,
```

Comment:
I know we're already looking into project id, but it looks like region can also be extracted from the authentication response. Lets kill ProviderClientOpts entirely as soon as we can.

Reply:
Sounds good to me, I never really understood why the `clientOpts` were necessary.

Reply:
Should probably be a follow up, though! Lets leave it alone in for this PR.